### PR TITLE
Rename to table_schema, column_schema, TableSchema

### LINKED ISSUE #774: Change file names to match new table-column schema structure
- Change `schema_column.py` file name to column_schema.py + testing files
- Change `Schema` class to `TableSchema` for better differentiation
  - might need to change some error wording
- Change schema.py to table_schema.py + relevant test files

### PR DESCRIPTION
**Pull request by tamargrey**
- In preparation for the ColumnSchema class, renames the schema and schema_column files to match `table_schema` and `column_schema` phrasing
- Updates various comments and docstrings to either use `TableSchema` when talking about the object or a more generic `schema` instead of `Schema`
- Closes #774

### CODE REVIEW

File: docs/source/release_notes.rst [hunk 1]
```diff
@@ -20,6 +20,7 @@ Release Notes
         * Remove ``ColumnNameMismatchWarning`` (:pr:`777`)
         * ``get_column_dict`` does not use standard tags by default (:pr:`782`)
         * Make ``logical_type`` and ``name`` params to ``_get_column_dict`` optional (:pr:`786`)
+        * Change schema and schema_column files and Schema object to match new table-column schema structure (:pr:`789`)
```
Comment (thehomebrewnerd) [inline docs/source/release_notes.rst:23]: How about a slight simplification - `Rename Schema object and files to match new table-column schema structure`?

**Review by thehomebrewnerd — COMMENTED** [review]
Looks good. We should probably add a breaking change note in the release notes as any prior code that referenced the `Schema` object will break with this change.

File: docs/source/release_notes.rst (hunk 1 above)
Comment (tamargrey) [reply docs/source/release_notes.rst]: This is much cleaner--changed!

**Review by thehomebrewnerd — APPROVED** [review]
Looks good!

**Pull request merged**